### PR TITLE
refactor: client syntax to match modern Gofer

### PR DESCRIPTION
**Pull request by aaarichter**
follow up to https://github.com/groupon/nlm/pull/57#discussion_r374845762

- [x] change client syntax to match modern Gofer

### CODE REVIEW

File: lib/github/client.js [hunk 1]
```diff
@@ -32,87 +32,101 @@
...
+const pkg = require('../../package');
+
 const parseRepository = require('./parse-repository');
 
-function Github(config) {
-  Gofer.call(this, config, 'github');
-}
+class Github extends Gofer {
+  constructor(config) {
+    super(config, 'github', pkg.version);
```
Comment (dbushong) [inline lib/github/client.js]:
```suggestion
    super(config, 'github', pkg.version, pkg.name);
```

(as long as we've got `pkg` handy anyway...)

File: lib/github/client.js [hunk 2]
```diff
@@ -32,87 +32,101 @@
...
-        }).json();
-      },
-      get: function get(pullId) {
-        return request('/repos/{owner}/{repo}/pulls/{pullId}', {
-          pathParams: {
-            pullId,
-          },
-        }).json();
-      },
+    this.pull = {
```
Comment (dbushong) [inline lib/github/client.js]: In order of preference for this refactor, I'd probably go with:

1. replacing this whole client with `@octokit/rest` (though this is the most work and would probably require changing the test mocks to use something like nock)
1. replacing the nested legacy-style calls with a flat set of methods, e.g. `pullGet`, `pullCommits`, etc; then they could just be methods declared on the class (medium amount of work, need to change call locations)
1. if you want to keep the hierarchical style, and not change the other code, you could do this slightly cleaner(?) hack, which at least gets all of the method declarations out of the constructor:

```js
class Github extends Gofer {
  //  ... constructor ...

  get pull() {
    return {
      get: pullId => this.get(/* ... */).json(),
      commits: pullId => this.get(/* ... */).json(),
      // ...
    };
  }

  get labels() {
    // ...
  }
}
```

Comment (aaarichter) [reply lib/github/client.js]: thanks for the suggestions. I'll go with number 3. force pushing onto the existing commit

File: lib/github/client.js [hunk 3]
```diff
@@ -32,87 +32,108 @@
...
-      create: function create(label) {
-        return request('/repos/{owner}/{repo}/labels', {
+      /**
+       *
+       * @param {string} label
+       * @return {Promise<any>}
+       */
+      create: label =>
+        this.get('/repos/{owner}/{repo}/labels', {
           method: 'POST',
```
Comment (dbushong) [inline lib/github/client.js]: `this.post()` ?

File: lib/github/client.js [hunk 4]
```diff
@@ -32,87 +32,108 @@
...
+        }).json(),
+
+      /**
+       * @param {string} issueId
+       * @param {string[]} labels
+       * @return {FetchResponse}
+       */
+      setForIssue: (issueId, labels) =>
+        this.get('/repos/{owner}/{repo}/issues/{issueId}/labels', {
           method: 'PUT',
```
Comment (dbushong) [inline lib/github/client.js]: `this.put()`

File: lib/github/client.js [hunk 5]
```diff
@@ -32,87 +32,108 @@
...
-          },
-        }).json();
-      },
+      /**
+       * @param {string} pullId
+       * @return {Promise<any>}
+       */
+      get: pullId =>
+        this.get('/repos/{owner}/{repo}/pulls/{pullId}', {
+          pathParams: { pullId },
```
Comment (dbushong) [inline lib/github/client.js]:
```suggestion
          pathParams: { pullId },
          endpointName: 'pull.get',
```

ideally, stick `endpointName` on all of these (one downside of the class syntax)

**Review by dbushong — APPROVED** [review]

**Pull request merged**